src/System.IO.Pipelines/src/System/IO/Pipelines/PipeReader.cs

```diff
@@ -75,16 +75,27 @@ public virtual Stream AsStream(bool leaveOpen = false)
         public abstract void CancelPendingRead();
 
         /// <summary>
-        /// Signal to the producer that the consumer is done reading.
+        /// Marks the <see cref="PipeReader"/> as being complete, meaning no more data will be read from it.
         /// </summary>
-        /// <param name="exception">Optional <see cref="Exception"/> indicating a failure that's causing the pipeline to complete.</param>
+        /// <param name="exception">Optional <see cref="Exception"/> indicating a failure that's causing the reader to complete.</param>
         public abstract void Complete(Exception exception = null);
 
+        /// <summary>
+        /// Marks the <see cref="PipeReader"/> as being complete, meaning no more data will be read from it.
+        /// </summary>
+        /// <param name="exception">Optional <see cref="Exception"/> indicating a failure that's causing the reader to complete.</param>
+        public virtual ValueTask CompleteAsync(Exception exception = null)
+        {
+            Complete(exception);
```

Comment:
Might Complete throw?  In such cases we generally try to have the exception emerge from the returned task, e.g.
https://github.com/dotnet/corefx/blob/9bf9ae013134c6ff88fd905bd13fe07da96b66b4/src/Common/src/CoreLib/System/IO/Stream.cs#L236-L247

Reply:
Hmm complete throwing would be bad but I can make this change regardless